addons/mail/controllers/discuss.py

```diff
@@ -570,3 +570,41 @@ def mail_guest_update_name(self, guest_id, name):
         if guest_to_rename_sudo != guest and not request.env.user._is_admin():
             raise NotFound()
         guest_to_rename_sudo._update_name(name)
+
+    # --------------------------------------------------------------------------
+    # Link preview API
+    # --------------------------------------------------------------------------
+
+    @http.route('/mail/link_preview', type='json', auth='public')
+    def link_preview(self, url, channel_id):
+        channel_partner = request.env['mail.channel.partner']
+        if request.env.user._is_public():
+            channel_partner = request.env['mail.channel.partner']._get_as_sudo_from_request_or_raise(request=request, channel_id=int(channel_id))
+        open_graph_data = request.env['ir.attachment']._get_data_from_url(url)
```

Comment:
Does it really have to be named `open_graph_data` knowing that the method can also return an image?
```suggestion
        link_preview_data = request.env['ir.attachment']._get_data_from_url(url)
```